crates/feldera-types/src/query_params.rs

```diff
@@ -0,0 +1,27 @@
+//! Types for the query parameters of the pipeline endpoints.
+
+use serde::Deserialize;
+use utoipa::{IntoParams, ToSchema};
+
+/// Circuit metrics output format.
+/// - `prometheus`: format expected by Prometheus, as documented at:
+///   <https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md>
+/// - `json`: JSON format
+#[derive(Debug, Deserialize, ToSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum MetricsFormat {
+    Prometheus,
+    Json,
+}
+
+/// Returns default metrics format.
+fn default_metrics_format() -> MetricsFormat {
+    MetricsFormat::Prometheus
+}
+
+/// Query parameters to retrieve pipeline circuit metrics.
+#[derive(Debug, Deserialize, IntoParams, ToSchema)]
+pub struct MetricsParameters {
+    #[serde(default = "default_metrics_format")]
```

Comment:
```suggestion
    #[serde(default)]
```

If you take this and the previous suggestion, then you can drop the default_metrics_format function.

Reply:
Thanks for the feedback, the PR is already merged, but will keep this in mind. It's indeed a lot shorter/clearer to express a default for an enumeration in that way.